### ISSUE TITLE
Spacebar Pause Doubling Pause

### ISSUE THREAD

**Issue by Layendan** [vidstack/player#788]
### Current Behavior:

When using safari an chromium *(Edge and Chrome, not tested on others)* fullscreen api pausing a video while in fullscreen pauses the video by default. The new keybinds calls `pause = !pause` again making the video play again.

### Expected Behavior:

Either `e.preventDefault()` or `e.stopPropagation()` *(Not Tested)*, 
or check for 
```ts
// Checks if fullscreen since spacebar pauses automatically
// Checks to see if using chromium since spacebar pauses automatically
if ( // @ts-ignore
    !document.webkitCurrentFullScreenElement &&
    !document.exitFullscreen ) {
        paused = !paused;
}
```
 to pause only when not in fullscreen when pressing spacebar.

### Steps To Reproduce:

*(Prereqs: Rust + Node + Yarn)*
1. Clone https://github.com/Layendan/NineAnimator-Tauri
2. Run `yarn install`
3. Run `yarn dev` *(Opens up a new window that uses Edge on Windows and Safari on MacOS)*
4. Click on any card
5. Click on an episode
6. Select fullscreen *(Optional for chrome, maybe edge)*
7. Pause using spacebar

This runs on `http://localhost:5173/` which allows you to follow steps 4-7 on any browser you wish.

### Environment:

- Framework: Svelte
- Meta Framework: SvelteKit
- Node: 19.6.0
- Device: Mac M1 2020, Windows Intel CPU
- OS: MacOS@13.1, Windows@(Replace when access to PC)
- Browser: Safari@16.2, Chrome@109.0.5414.119, Edge@(Replace when access to PC)

### Anything Else?

https://user-images.githubusercontent.com/46062388/222946744-9d4584b6-117d-426b-a2f8-de4f72d0b800.mp4

After hitting play with the mouse, I press spacebar multiple times to pause, but the video pauses and unpauses.

**Comment by Layendan** [comment]
I added more info about this in this discord thread: https://discord.com/channels/742612686679965696/1082046652992467024

**Comment by mihar-22** [comment]
For now I'd recommend disabling keyboard shortcuts if you're using native controls like so:

```html
<media-player key-disabled>
```

I think we'll just automatically disable it if `controls` is set.

**Comment by Layendan** [comment]
The problem for that is if the user is in inline video mode, they will have no controls and will be missing a few controls in fullscreen mode since WebKit does not inject that many shortcuts.

As mentioned previously, this pr https://github.com/vidstack/player/pull/789 fixes the aforementioned problem. Allowing for vidstack's custome keyboard shortcuts without interfering with the default shortcuts.

**Comment by mihar-22** [comment]
We can't blanket stop propagrating all trigger events because they don't belong to us, they could be provided by the user and we shouldn't stop UI events either. 

I see what you're saying. The keyboard shortcuts weren't intended for use with native controls but I've also noticied native shortcuts are very limited. I'll have another review of this and see what we can do, tricky part is we can't tell which shortcuts are supported by which browser without manual testing and that could change any day.

**Comment by Layendan** [comment]
> We can't blanket stop propagation of all trigger events because they don't belong to us, they could be provided by the user and we shouldn't stop UI events either. 

I see what you mean. That's something I haven't thought of. Sorry for pushing a faulty fix so much.

> we can't tell which shortcuts are supported by which browser without manual testing and that could change any day.

If you need any help on this, you can contact me on discord and I'll try and help
